x/storage/keeper/msg_server.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/CosmWasm/wasmd/x/wasm/ioutils"
+	"github.com/hyperledger/burrow/crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/x/storage/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) Store(goCtx context.Context, msg *types.MsgStore) (*types.MsgStoreResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	data := msg.Data
+
+	var err error
+	if ioutils.IsGzip(data) {
+		data, err = ioutils.Uncompress(data, int64(800*1024))
```

Comment:
Just to be sure. We are not susceptible to zip bombing?

Reply:
Thanks for the heads up. Will make sure of this in the new PR that will contain full implemenetation of the module.